PrepareFiles.py

```diff
@@ -1,106 +1,164 @@
+    client = boto3.client('sts')
+    sts_response = client.assume_role(
+        RoleArn=role_arn,
+        RoleSessionName='cross_acct_lambda',
+        DurationSeconds=900
+    )
+    ACCESS_KEY = sts_response['Credentials']['AccessKeyId']
+    SECRET_KEY = sts_response['Credentials']['SecretAccessKey']
+    SESSION_TOKEN = sts_response['Credentials']['SessionToken']
+    s3_client = boto3.client(
+        's3',
+        aws_access_key_id=ACCESS_KEY,
+        aws_secret_access_key=SECRET_KEY,
+        aws_session_token=SESSION_TOKEN
+    )
 else:
-    s3_bucket_put = PreparedFiles_config.S3_BUCKET_NAME_PUT
-if "Digital_MetaData_API" in os.environ:
-    metadata_api = os.environ['Digital_MetaData_API']
-else:
-    metadata_api = PreparedFiles_config.Digital_MetaData_API
+    s3_bucket_get = 0
+    s3_bucket_put = 0
+    metadata_api = 0
+    role_arn = 0
 
 
 def get_replica(rid):
-    with urllib.request.urlopen(metadata_api+rid) as content_file:
+    with urllib.request.urlopen(metadata_api + 'replicas/' + rid) as content_file:
         json_content = content_file.read()
     content = json.loads(json_content)
     replica = Replica(content)
     return replica
 
 
 class Replica:
-    def __init__( self, content ):
+    def __init__(self, content):
         self.replica_data = content
 
-    def process_files(self, delivery_type, max_deliveryfile_size, reference):
-        # TODO Code the ZIP case. Can any of the PDF case be generalised?
+    def process_files(self, delivery_type, max_deliveryfile_size, reference, iaid):
         status = False
         if delivery_type == "pdf":
-            status = self._create_pdf(max_deliveryfile_size, reference)
-        # else:
-        #     create_zip(cal_avg_size)
+            status = self._create_pdf(max_deliveryfile_size, reference, iaid)
+        else:
+            status = self._create_zip(max_deliveryfile_size, reference, iaid)
         return status
 
-    def _create_image_list(self, max_deliveryfile_size, data):
+    def _create_file_list(self, max_deliveryfile_size, data):
         output_file_list = []
         size = 0
-        this_pdf = []
+        this_file = []
         length = len(data)
         count = 1
         for file in data:
-            this_pdf.append(file['name'])
-            size = file['size'] + size
-            if size > max_deliveryfile_size:
-                output_file_list.append(this_pdf)
+            if length == count:
+                this_file.append(file['name'])
+                output_file_list.append(this_file)
+            elif (file['size'] + data[count]['size']) > max_deliveryfile_size:
+                this_file.append(file['name'])
+                output_file_list.append(this_file)
                 size = 0
-                this_pdf = []
-            elif length == count:
-                output_file_list.append(this_pdf)
+                this_file = []
+            else:
+                this_file.append(file['name'])
```

Comment:
Also this else case looks like it's doing nearly the same thing?

Reply:
Yeah, I think you're right. We probably don't need the below if statement anymore...